### PR TITLE
 SUS-3496 Use task queue to update dataware.pages 

### PR DESCRIPTION
**Pull request by mszabo-wikia**
Update global aggregate table of article pages using background task

https://wikia-inc.atlassian.net/browse/SUS-3496
 
Merge https://github.com/Wikia/backend/pull/312 when this one is deployed.

### CODE REVIEW

File: extensions/wikia/Pages/PagesHooks.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		static::scheduleArticleUpdate( $newId, $title->isContentPage() );
+	}
+
+	private static function scheduleArticleUpdate( int $articleId, bool $isContentPage ) {
+		if ( $articleId ) {
+			$task = UpdatePagesTask::newLocalTask();
+
+			$task->call( 'insertOrUpdateEntry', $articleId, $isContentPage );
+			$task->queue();
+		}
```
Comment (macbre) [inline extensions/wikia/Pages/PagesHooks.php]: Should we log cases when an `if` statement does not evaluate to true?

File: extensions/wikia/Pages/UpdatePagesTask.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+				'page_title',
+				'page_is_redirect',
+				'page_latest',
+				'rev_timestamp AS page_last_edited',
+			],
+			[ 'page_id' => $articleId ],
+			__METHOD__,
+			[],
+			[ 'revision' => [ 'INNER JOIN', [ 'page_latest = rev_id' ] ] ]
+		);
```
Comment (macbre) [inline extensions/wikia/Pages/UpdatePagesTask.php]: We risk a slave lag here. Can we pass this information as a task parameters and get them when the task is queued?

File: includes/wikia/DefaultSettings.php [hunk 3]
```diff
@@ -1902,3 +1902,6 @@
 
 // SUS-3475: Extension to update shared city_list table
 require_once "$IP/extensions/wikia/CityList/CityList.setup.php";
+
+// Extension to update shared dataware.pages table
```
Comment (macbre) [inline includes/wikia/DefaultSettings.php]: Let's add a ticket ID here :)

**Comment by ghost** [comment]
I would reconsider creating `trait`s.  While it is a good practice from the re-usability perspective, developers must be aware of the `trait`s existence in order to use it.  Apart from core MediaWiki classes and functions, I do not think there is any other API all developers are equally aware of.  We will likely end up with bunch of `trait`s and/or `abstract`s used once.

**Comment by mszabo-wikia** [comment]
@mixoteric Developers equally don't use the integration tests infrastructure either...

File: extensions/wikia/Pages/UpdatePagesTask.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+	/**
+	 * Construct a new {@see UpdatePagesTask} instance that will be run in the context of the current wiki
+	 * @return UpdatePagesTask
+	 */
+	public static function newLocalTask(): UpdatePagesTask {
+		global $wgCityId;
+
+		return ( new self() )->wikiId( $wgCityId );
+	}
```
Comment (macbre) [inline extensions/wikia/Pages/UpdatePagesTask.php:65]: Let's move this method to `BaseTask` class.

**Review by macbre — APPROVED** [review]

**Comment by macbre** [comment]
```
Function: DatabaseBase::sourceFile( /var/lib/jenkins/workspace/integrationTests_php_branches/app/extensions/wikia/UserRenameTool/tests/fixtures/dataware-pages.sql )
Error: 1 table `pages` already exists
 in /var/lib/jenkins/workspace/integrationTests_php_branches/app/includes/db/Database.php:1082
```

However, it runs fine on my devbox:

```
Running RenameUserPagesTaskTest ..                          101 / 101 (100%)done in 1201.90 ms [2 tests/0 skipped/0 incomplete]
```

**Pull request merged**